Tools/clinic/clinic.py

```diff
@@ -4906,22 +4906,26 @@ def bad_node(self, node):
         key = f"{parameter_name}_as_{c_name}" if c_name else parameter_name
         self.function.parameters[key] = p
 
-    def parse_converter(self, annotation):
-        if (isinstance(annotation, ast.Constant) and
-            type(annotation.value) is str):
-            return annotation.value, True, {}
+    KwargDict = dict[str | None, Any]
 
-        if isinstance(annotation, ast.Name):
-            return annotation.id, False, {}
-
-        if not isinstance(annotation, ast.Call):
-            fail("Annotations must be either a name, a function call, or a string.")
-
-        name = annotation.func.id
-        symbols = globals()
-
-        kwargs = {node.arg: eval_ast_expr(node.value, symbols) for node in annotation.keywords}
-        return name, False, kwargs
+    @staticmethod
+    def parse_converter(annotation: ast.expr | None) -> tuple[str, bool, KwargDict]:
+        match annotation:
+            case ast.Constant(value=str() as value):
+                return value, True, {}
```

Comment:
This branch is covered here:

https://github.com/python/cpython/blob/60993ba8b4a04dfcf2d3067d919bcce160f01bd5/Lib/test/test_clinic.py#L772-L775

_But_ there doesn't seem to be coverage for the case where an ast.Constant node is encountered that has a non-`str` value.

Reply:
Ok, can you add a test for that case?

Reply:
Same answer as https://github.com/python/cpython/pull/104696#discussion_r1199731051 :)